system/lib/libc/musl/src/stat/fstatat.c

```diff
@@ -1,10 +1,160 @@
+#define _BSD_SOURCE
 #include <sys/stat.h>
+#include <string.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <stdint.h>
+#include <sys/sysmacros.h>
 #include "syscall.h"
-#include "libc.h"
+#include "kstat.h"
 
-int fstatat(int fd, const char *restrict path, struct stat *restrict buf, int flag)
+/* XXX Emscripten: we ensure kstat == stat so we can simply make the syscall
```

Comment:
Can we add a static assert that the struct sizes match at least? Might avoid some future bugs.

Reply:
Updated the comment.  We literally define one to be other.